presto-main/src/main/java/com/facebook/presto/server/protocol/QueryResourceUtil.java

```diff
@@ -120,6 +127,31 @@ public static Response toResponse(Query query, QueryResults queryResults, boolea
         return response.build();
     }
 
+    public static Response toResponse(Query query, QueryResults queryResults, String xPrestoPrefixUri, boolean compressionEnabled)
```

Comment:
This method mutates the queryResults parameter as part of the method's body. I wondered if this could cause unexpected code behaviors in other codes that might depend on this method. From the method's name, it might not be clear that such mutation occurs. If that makes sense, we could build a new queryResults object and mutate it as part of the result.

Reply:
Yeah good catch. The queryResults objects are actually stored in a cache and may be served multiple times.
I've now modified the code to create a clone of the object and modify that clone.